sdcm/stress_thread.py

```diff
@@ -145,43 +134,67 @@ def _add_errors_option(stress_cmd: str, to_add: list) -> str:
             return stress_cmd
         return stress_cmd.replace(current_error_option, 'errors ' + ' '.join(new_error_suboptions))
 
-    def _get_available_suboptions(self, node, option, _cache={}):  # pylint: disable=dangerous-default-value
-        if cached_value := _cache.get(node.name):
+    def _get_available_suboptions(self, loader, option, _cache={}):  # pylint: disable=dangerous-default-value
+        if cached_value := _cache.get(option):
             return cached_value
         try:
-            result = node.remoter.run(
+            result = loader.run(
                 cmd=f'cassandra-stress help {option} | grep "^Usage:"',
                 timeout=self.timeout,
                 ignore_status=True).stdout
         except Exception:  # pylint: disable=broad-except
             return []
         findings = re.findall(r' *\[([\w-]+?)[=?]*] *', result)
-        _cache[node.name] = findings
+        _cache[option] = findings
         return findings
 
     @staticmethod
     def _disable_logging_for_cs(node, _cache={}):  # pylint: disable=dangerous-default-value
         if not (node.is_kubernetes() or node.name in _cache):
-            node.remoter.run("cp /etc/scylla/cassandra/logback-tools.xml .", ignore_status=True)
+            node.run("cp /etc/scylla/cassandra/logback-tools.xml .", ignore_status=True)
             _cache[node.name] = 'done'
 
-    def _run_stress(self, node, loader_idx, cpu_idx, keyspace_idx):  # pylint: disable=too-many-locals
-        stress_cmd = self.create_stress_cmd(node, keyspace_idx)
+    def _run_stress(self, loader, loader_idx, cpu_idx):
+        pass
+
+    def _run_cs_stress(self, loader, loader_idx, cpu_idx, keyspace_idx):  # pylint: disable=too-many-locals
+        cpu_options = ""
+        if self.stress_num > 1:
+            cpu_options = f'--cpuset-cpus="{cpu_idx}"'
+        cmd_runner_name = loader.ip_address
```

Comment:
`loader.ip_address` must not even called in case of `dynamic` loaders. Need to put it in the conditional branch of code.

Reply:
fixed